### PR TITLE
Adding Hcsr04 with RMT for ESP32

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Adding Hcsr04 with RMT for ESP32

## Motivation and Context

- Adding Hcsr04 with RMT for ESP32
- More precise than the generic Hcsr04 implementation

## How Has This Been Tested?

On a real device.

## Screenshots
image

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: devices/Hcsr04.Esp32/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## Documentation
+
+* [HC-SR04 data sheet](https://components101.com/sites/default/files/component_datasheet/HCSR04%20Datasheet.pdf)
+
+## Usage
+
+In most of the cases, you just need to pass the 2 pins you're going to use: the Echo and Trigger pins.
+
+```sharp
```
Comment (josesimoes) [inline devices/Hcsr04.Esp32/README.md:13]:
```suggestion
```csharp
```

File: devices/Hcsr04.Esp32/Hcsr04.Esp32.nfproj [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  <ItemGroup>
+    <None Include="packages.config" />
+  </ItemGroup>
+  <Import Project="$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets" Condition="Exists('$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets')" />
+  <ProjectExtensions>
+    <ProjectCapabilities>
+      <ProjectConfigurationsDeclaredAsItems />
+    </ProjectCapabilities>
+  </ProjectExtensions>
+</Project>
```
Comment (josesimoes) [inline devices/Hcsr04.Esp32/Hcsr04.Esp32.nfproj:52]: Missing the entries for nbgv.

File: devices/Hcsr04.Esp32/packages.config [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="nanoFramework.CoreLibrary" version="1.10.5" targetFramework="netnanoframework10" />
+  <package id="nanoFramework.Hardware.Esp32.Rmt" version="1.2.1-preview.7" targetFramework="netnanoframework10" />
+  <package id="nanoFramework.Runtime.Events" version="1.9.2-preview.8" targetFramework="netnanoframework10" />
+  <package id="nanoFramework.System.Device.Model" version="1.0.176" targetFramework="netnanoframework10" />
+  <package id="UnitsNet.nanoFramework.Length" version="4.104.0" targetFramework="netnanoframework10" />
+</packages>
```
Comment (josesimoes) [inline devices/Hcsr04.Esp32/packages.config:8]: Missing nbgv

**Review by josesimoes — COMMENTED** [review]
All good! Just missing the ref to nbgv. 💪😉

File: devices/Hcsr04.Esp32/Hcsr04.Esp32.nfproj (hunk 2 above)
Comment (Ellerbach) [reply devices/Hcsr04.Esp32/Hcsr04.Esp32.nfproj:59]: I knew I forgot something :-D Great catch, thanks.

**Review by josesimoes — APPROVED** [review]
Perfect! 👍🏻

**Pull request merged**